### ISSUE TITLE
Explore using Succinct Data for querying on compressed data

### ISSUE THREAD

**Issue by thedrow** [citusdata/citus#5311]
I saw your release announcement on 10.2 (congrats 🎉) and I was wondering if querying indexes always required decompressing at least one chunk.
It turns out there's a way that can query on compressed data. The researchers dubbed them Succinct Indexes.
The [paper can be found here](https://www.usenix.org/conference/nsdi15/technical-sessions/presentation/agarwal).

The Java [reference implementation](https://github.com/amplab/succinct/blob/master/core/README.md) is in Github, of course.

**Comment by onurctirtir** [comment]
> if querying indexes always required decompressing at least one chunk.

Yes. In the current design, columnar decompresses one chunk group for each row to be fetched from columnar table during index scan, unless the chunk group was already decompressed when fetching the previous row.

Thanks a lot @thedrow for sharing this paper ! I will try to spend some time on that soon.

**Comment by jeff-davis** [comment]
@thedrow Thank you for sharing the paper.

I'm worried that this would be a fairly large expansion of the scope and complexity. I'll see what @onurctirtir has to say after he's investigated.
